rust/src/smb/smb.rs

```diff
@@ -53,6 +53,23 @@ use crate::smb::events::*;
 use crate::smb::files::*;
 use crate::smb::smb2_ioctl::*;
 
+#[repr(C)]
+pub enum SMBPduType {
```

Comment:
on question about the implementation is how much granularity we need to add wrt to the record types. Does it make sense to split by SMB version here?